erlang/api_cowboy/apps/api_cowboy/src/api_cowboy_handler.erl

```diff
@@ -0,0 +1,145 @@
+    Json = jsx:encode(ItemsWithKeys),
+
+    Req = cowboy_req:reply(200,
+        #{<<"content-type">> => <<"application/json">>},
+        Json,
+        Req0),
+    {ok, Req, State}.
+
+handle_get_item(Req0, State, ItemName) ->
+    {ok, Client} = connect_to_redis(),
+    {ok, Value} = eredis:q(Client, ["GET", ItemName]),
+    case Value of
+        undefined ->
+            Req = cowboy_req:reply(404,
+                #{<<"content-type">> => <<"text/plain">>},
+                <<"Item not found">>,
+                Req0),
+            {ok, Req, State};
+        _ ->
+            Req = cowboy_req:reply(200,
+                #{<<"content-type">> => <<"application/json">>},
+                Value,
+                Req0),
+            {ok, Req, State}
+    end.
+
+handle_post_items(Req0, State) ->
+    {ok, Body, _} = cowboy_req:read_body(Req0),
+    ParsedList = jsx:decode(Body),
+    ParsedBody = maps:from_list(ParsedList),
+
+    Name = maps:get(<<"name">>, ParsedBody),
+    {ok, Client} = connect_to_redis(),
+    {ok, <<"OK">>} = eredis:q(Client, ["SET", Name, jsx:encode(ParsedBody)]),
+
+    Result = <<"Item ", Name/binary, " created successfully">>,
+    Req = cowboy_req:reply(201, #{<<"content-type">> => <<"application/json">>},
+        Result,
+        Req0),
+    {ok, Req, State}.
+
+handle_put_item(Req0, State, ItemName) ->
+    {ok, Body, _} = cowboy_req:read_body(Req0),
+    ParsedList = jsx:decode(Body),
+    ParsedBody = maps:from_list(ParsedList),
+
+    {ok, Client} = connect_to_redis(),
+    {ok, <<"OK">>} = eredis:q(Client, ["SET", ItemName, jsx:encode(ParsedBody)]),
+    Result = <<"Item ", ItemName/binary, " updated successfully">>,
+
+    Req = cowboy_req:reply(200, #{<<"content-type">> => <<"application/json">>},
+        Result,
+        Req0),
+    {ok, Req, State}.
+
+handle_patch_item_price(Req0, State, ItemName) ->
+    {ok, Body, _} = cowboy_req:read_body(Req0),
+    ParsedList = jsx:decode(Body),
+    ParsedBody = maps:from_list(ParsedList),
+
+    {ok, Client} = connect_to_redis(),
+    {ok, Value} = eredis:q(Client, ["GET", ItemName]),
+
+    CurrentItem = maps:from_list(jsx:decode(Value)),
+    NewPrice = maps:get(<<"price">>, ParsedBody, maps:get(<<"price">>, CurrentItem)),
+    UpdatedItem = maps:put(<<"price">>, NewPrice, CurrentItem),
+
+    {ok, <<"OK">>} = eredis:q(Client, ["SET", ItemName, jsx:encode(UpdatedItem)]),
+    Result = <<"Item ", ItemName/binary, " updated successfully">>,
+
+    Req = cowboy_req:reply(200, #{<<"content-type">> => <<"application/json">>},
+        Result,
+        Req0),
+    {ok, Req, State}.
+
+handle_delete_item(Req0, State, ItemName) ->
+    {ok, Client} = connect_to_redis(),
+    {ok, DeletedCount} = eredis:q(Client, ["DEL", ItemName]),
+    Result = case DeletedCount of
+        <<"0">> -> <<"Item ", ItemName/binary, " not found">>;
+        <<"1">> -> <<"Item ", ItemName/binary, " deleted successfully">>;
+        _ -> <<"Unexpected result: ", DeletedCount/binary>>
```

Comment:
I recommend this logic to be separated from the handler as the handler shall only handle the http request, not the logic of your domain itself